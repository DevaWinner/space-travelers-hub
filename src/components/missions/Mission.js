import React from 'react';
import PropTypes from 'prop-types';

const Mission = ({ mission, handleJoinMission, handleLeaveMission }) => (
  <tr>
    <td className="mission-name">{mission.mission_name}</td>
    <td className="mission-desc">{mission.description}</td>
    <td className="status">
      {mission.reserved ? (
        <span className="active-member-badge" type="button">Active Member</span>
      ) : (
        <span className="inactive-member-badge" type="button">Not a Member</span>
      )}
    </td>
    <td className="btn-td">
      {mission.reserved ? (
        <button className="leave-button" onClick={() => handleLeaveMission(mission.mission_id)} type="button">Leave Mission</button>
      ) : (
        <button className="join-button" onClick={() => handleJoinMission(mission.mission_id)} type="button">Join Mission</button>
      )}
    </td>
  </tr>
);

Mission.propTypes = {
  mission: PropTypes.shape({
    mission_id: PropTypes.string.isRequired,
    mission_name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    reserved: PropTypes.bool,
  }).isRequired,
  handleJoinMission: PropTypes.func.isRequired,
  handleLeaveMission: PropTypes.func.isRequired,
};
export default Mission;
