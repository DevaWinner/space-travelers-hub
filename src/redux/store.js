import rocketsSlice from './rockets/rocketsSlice';
