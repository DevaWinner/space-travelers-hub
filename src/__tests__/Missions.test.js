import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import Mission from '../components/missions/Mission';
import store from '../redux/store';

const render = (component) => rtlRender(
  <Provider store={store}>
    {component}
  </Provider>,
);

describe('My Missions', () => {
  it('Render Mission Components', () => {
    const mission = {
      mission_id: '1',
      mission_name: 'Sample Mission',
      description: 'This is a sample mission',
      reserved: true,
    };

    render(<Mission mission={mission} />);
    expect(Mission).toMatchSnapshot();
  });
});
