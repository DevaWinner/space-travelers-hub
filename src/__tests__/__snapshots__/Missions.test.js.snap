// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Missions Render Mission Components 1`] = `[Function]`;
